src/models/training/t5_training_params.py

```diff
@@ -0,0 +1,141 @@
+from transformers import T5Tokenizer, Seq2SeqTrainingArguments
+
+from src.data.science_qa_dataset_img import ScienceQADatasetImg, img_shape
+from src.data.science_qa_dataset_std import ScienceQADatasetStd
+from src.models.model import T5ForConditionalGeneration, T5ForMultimodalGeneration
+
+
+def get_t5_model(args, tokenizer: T5Tokenizer, save_dir: str):
+    if is_img_type_knonw(args):
+        padding_idx = tokenizer._convert_token_to_id(tokenizer.pad_token)
+        patch_size = img_shape[args.img_type]
+        model = T5ForMultimodalGeneration.from_pretrained(
+            args.model, patch_size=patch_size, padding_idx=padding_idx, save_dir=save_dir)
+    else:
+        model = T5ForConditionalGeneration.from_pretrained(args.model)
+    return model
+
+
+def get_training_data(args, dataframe, tokenizer):
+
+    problems = dataframe['problems']
+    qids = dataframe['qids']
+    train_qids = qids['train']
+    test_qids = qids['test']
+    val_qids = qids['val']
+
+    if is_img_type_knonw(args):
+        name_maps = dataframe['name_maps']
+        image_features = dataframe['image_features']
+
+        train_set = ScienceQADatasetImg(
+            problems,
+            train_qids,
+            name_maps,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+            image_features,
+        )
+        eval_set = ScienceQADatasetImg(
+            problems,
+            val_qids,
+            name_maps,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+            image_features,
+            args.eval_le,
+        )
+        test_set = ScienceQADatasetImg(
+            problems,
+            test_qids,
+            name_maps,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+            image_features,
+            args.test_le,
+        )
+    else:
+        train_set = ScienceQADatasetStd(
+            problems,
+            train_qids,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+        )
+        eval_set = ScienceQADatasetStd(
+            problems,
+            val_qids,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+            args.eval_le,
+        )
+
+        test_set = ScienceQADatasetStd(
+            problems,
+            test_qids,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+            args.test_le,
+        )
+
+    return train_set, eval_set, test_set
```

Comment:
Let's make a factory for the ScienceQADataset. It can be done in a next iteration, but should be easy to fit into this